ONSdigital/dp-api-clients-go
# Feature/client health checker

### What

Add checker functions for current clients:
* codelist
* dataset
* filter
* hierarchy
* importapi
* search
* zebedee

Updated health package to handle more scenarios, update client packages to use latest version of the dp-rchttp library and switch out `go-ns/log` for  `log.go/log` lib. Implement go modules for dependency management and also add documentation for use of health package.

### How to review

* Check changes make sense and are logical
* Check unit tests pass (and that it builds dependencies from go.mod)
* Check all clients in library have Checker function and abides to [healthcheck spec](https://github.com/ONSdigital/dp/blob/master/standards/HEALTH_CHECK_SPECIFICATION.md)

### Who can review

Anyone
